source/api/galaxy/DatasetGenerationTool.py

```diff
@@ -41,7 +41,9 @@ def update_specs(self):
         with open(self.yaml_path, 'r') as file:
             specs = yaml.safe_load(file)
 
-        ParameterValidator.assert_keys(specs.keys(), ['definitions', 'instructions'], DatasetGenerationTool.__name__, 'YAML specification')
```

Comment:
there is a function assert keys present in ParameterValidator that does this (not exclusive for the given keys). Also if allowing output, it should be in the error message as well.

Reply:
ok, I had kind of intentionally done it like this so the user wouldn't be confused. We always want definitions and instructions (missing one of those two would also be bad), but don't need to confuse the user by saying they need to add 'output' if they are only missing definitions or instructions, and don't need to require them to remove 'output' if it was already in there (downloaded files from galaxy)

Reply:
ok :)

Reply:
updated it now to a more elegant and complete solution with ParameterValidator